ios/FluentUI/Navigation/UINavigationItem+Navigation.swift

```diff
@@ -101,12 +101,23 @@ import UIKit
         }
     }
 
+    var titleStyle: NavigationBar.TitleStyle {
+        get {
+            return objc_getAssociatedObject(self, &AssociatedKeys.titleStyle) as? NavigationBar.TitleStyle ?? .system
+        }
+        set {
+            objc_setAssociatedObject(self, &AssociatedKeys.titleStyle, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+        }
+    }
+
     var usesLargeTitle: Bool {
         get {
-            return objc_getAssociatedObject(self, &AssociatedKeys.usesLargeTitle) as? Bool ?? false
+            assertionFailure("`usesLargeTitle` will be deprecated in a future release. Use `titleStyle` instead.")
```

Comment:
If the goal is to use an assert instead of a deprecation message, you should probably use just `assert` instead of `assertionFailure`. I'm not sure them trying to use `usesLargeTitle` is worth crashing over, especially in release builds.

Reply:
`assertionFailure` doesn't crash on release builds anyway.

https://www.hackingwithswift.com/plus/intermediate-swift/understanding-assertions#assertionfailure

Reply:
Ah, so it doesn't. That's misleading of the documentation. Explicitly calling out that `assert` is just in debug, and then saying nothing about `assertionFailure`. I guess debug vs release is `assert` vs `precondition`. Regardless, we probably don't want to crash on our debug builds, either.

Reply:
Discussed offline. Let's just outright deprecate `usesLargeTitle` to help nudge consumers of this library to update to the newest API.